### PR TITLE
design test cases for test_config.py

### PR DESCRIPTION
**Pull request by Junefen**
**tests/conftest**
- New add fixtures `function_unregister_host` `class_unregister_host` `rhsmconf` `function_rhsmconf_recovery`
- Redefine the fixture names to started with `function_` or `class_`

**base.py**
- move the `msg_search()` and `msg_number()` to base.py from runner.py to make them public.

**configure.py**
- add new class `RHSMConf` to maintain the /etc/rhsm/rhsm.conf
- include `update ()` and `recovery()` functions

**test_rhsm.py**
positive cases (tier1)
- test_no_rhsm_options
- test_rhsm_encrypted_password

negative cases (tier2)
- test_owner
- test_rhsm_hostname
- test_rhsm_port
- test_rhsm_prefix
-test_rhsm_username
- test_rhsm_password
- test_rhsm_encrypted_password

**Test Result**
```
 % python3 -m pytest --disable-warnings -v tests/subscription/test_rhsm.py                           
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 9 items                                                                                                                     

tests/subscription/test_rhsm.py::TestRHSMPositive::test_no_rhsm_options PASSED                                                  [ 11%]
tests/subscription/test_rhsm.py::TestRHSMPositive::test_rhsm_encrypted_password PASSED                                          [ 22%]
tests/subscription/test_rhsm.py::TestRHSMNegative::test_owner PASSED                                                            [ 33%]
tests/subscription/test_rhsm.py::TestRHSMNegative::test_rhsm_hostname PASSED                                                    [ 44%]
tests/subscription/test_rhsm.py::TestRHSMNegative::test_rhsm_port PASSED                                                        [ 55%]
tests/subscription/test_rhsm.py::TestRHSMNegative::test_rhsm_prefix PASSED                                                      [ 66%]
tests/subscription/test_rhsm.py::TestRHSMNegative::test_rhsm_username PASSED                                                    [ 77%]
tests/subscription/test_rhsm.py::TestRHSMNegative::test_rhsm_password PASSED                                                    [ 88%]
tests/subscription/test_rhsm.py::TestRHSMNegative::test_rhsm_encrypted_password PASSED                                          [100%]

============================================= 9 passed, 27 warnings in 1123.30s (0:18:43) =============================================
```

### CODE REVIEW

File: tests/subscription/test_rhsm.py [hunk 1]
```diff
@@ -3,42 +3,4 @@
...
-        :tags: tier2
-        :customerscenario: false
-        :upstream: no
-        :steps:
-            1.
-        :expectedresults:
-            1.
-        """
-        logger.info("Succeeded to run the 'test_vdc_remove'")
+"""
```
Comment (hkx303) [inline tests/subscription/test_rhsm.py:6]: We have the Prompt here: No newline at end of file

File: tests/subscription/test_satellite.py [hunk 2]
```diff
@@ -3,42 +3,4 @@
...
-        :tags: tier2
-        :customerscenario: false
-        :upstream: no
-        :steps:
-            1.
-        :expectedresults:
-            1.
-        """
-        logger.info("Succeeded to run the 'test_vdc_remove'")
+"""
```
Comment (hkx303) [inline tests/subscription/test_satellite.py:6]: We have the Prompt here: No newline at end of file

**Review by hkx303 — APPROVED** [review]
ACK with a small comment, thanks for working on the great PR

File: tests/subscription/test_rhsm.py (hunk 1 above)
Comment (Junefen) [reply tests/subscription/test_rhsm.py:6]: thanks, will update it in the next pr

**Pull request merged**